### PR TITLE
Add the time an activity sign-up opens to the activities page

### LINKED ISSUE #1100: Feat: Activity sign up time
# Idea

There is currently no indication for members when the sign up opens for an activity.
When creating the activity this field is present, we should probably show this on the activity for members.

### PR DESCRIPTION
**Pull request by TobiasDeBruijn**
Fixes #1100

### CODE REVIEW

File: app/views/members/activities/partials/_view.html.haml [hunk 1]
```diff
@@ -45,6 +45,12 @@
           \-
           = activity.end_time.strftime("%H:%M")
       <br/>
+      - if activity.open_date != nil && activity.open_date > Date.today
```
Comment (SanderHageman) [inline app/views/members/activities/partials/_view.html.haml:48]: A quick google suggests to me that `Date.today` in rails shouldn't really be used, [please validate](http://winstonyw.com/2014/03/03/time_now_vs_time_zone_now/). I haven't checked which time/function is used for the check to see if someone can register for an activity but these need to be the same and I would suggest creating a helper function in ruby

File: app/views/members/activities/partials/_view.html.haml [hunk 2]
```diff
@@ -45,6 +45,12 @@
...
+        %strong= I18n.t("activerecord.labels.activities.opens_at")
+        = l(activity.open_date, format: "%A %d/%m/%Y")
+        - if !activity.open_time.nil?
```
Comment (SanderHageman) [inline app/views/members/activities/partials/_view.html.haml:51]: I'm unsure about the interactivity between ruby and haml but the activity model also contains a `when_open` method which includes both time and date. Current logic suggests to me that the open date will be visible for the whole day the activity opens; is that intended?

**Review by SanderHageman — CHANGES_REQUESTED** [review]
Some questions

File: app/views/members/activities/partials/_view.html.haml (hunk 1 above)
Comment (TobiasDeBruijn) [reply app/views/members/activities/partials/_view.html.haml:48]: We use `Date.today` everywhere for activities, so I think it'd be best to stick with that?

File: app/views/members/activities/partials/_view.html.haml (hunk 2 above)
Comment (TobiasDeBruijn) [reply app/views/members/activities/partials/_view.html.haml:51]: Good catch, it is not.

Looking into it further, there is actually a function on Activity: `open` (`:: bool`), so I'll just use the inverse of that.

File: config/locales/en.yml [hunk 3]
```diff
@@ -382,6 +382,7 @@ en:
     labels:
       activities:
         date: Date
+        opens_at: Signup from
```
Comment (SilasPeters) [inline config/locales/en.yml:385]: Although 'Signup from' is fine, I would like to propose two alternatives to see what you prefer.

"Enrollable from", the confirmation message when you signed up for an activity says "Succesfully _enrolled_ for this activity" and this is more consistent with that. I am not certain whether this is better.

"(Admissions) Open(s) at", not for a particular reason, just reads better imo.

**Review by SilasPeters — CHANGES_REQUESTED** [review]
I like these changes, and they are all idiomatic. However I suggest that when an activity will be enrollable at some point in the future, that the 'enroll' button is not hidden but grayed out. This better conveys the message of "you can/must enroll to participate!"

This can be another PR, but I suggest we do it now before we forget. Hence the "requested changes."

File: config/locales/en.yml (hunk 3 above)
Comment (SilasPeters) [reply config/locales/en.yml:385]: Also, shouldn't "open_at" be "open_from"? I don't know if that is more idiomatic

File: app/views/members/activities/partials/_view.html.haml (hunk 1 above)
Comment (SanderHageman) [reply app/views/members/activities/partials/_view.html.haml]: consistency 🫠

**Review by SanderHageman — DISMISSED** [review]

File: config/locales/en.yml (hunk 3 above)
Comment (TobiasDeBruijn) [reply config/locales/en.yml:385]: Good point. I'll change the key to `open_from` and change the text to `Enrollable from`.

The button will be another PR, it isn't relevant here :)

**Review by SilasPeters — APPROVED** [review]
Ready to roll

**Pull request merged**